lib/decorators/continuation.js

```diff
@@ -0,0 +1,60 @@
+const Cacheable = require('./cacheable');
+const { ref } = require('objection');
+const { get, isUndefined } = require('lodash');
+
+module.exports = settings => {
+
+  const { Project } = settings.models;
+  const cache = Cacheable();
+
+  return c => {
+    const model = get(c, 'data.model');
+    const action = get(c, 'data.action');
+
+    if (model === 'project' && (action === 'grant' || action === 'transfer')) {
+      const continuation = get(c, 'data.continuation');
+      // continuation added on init, noop.
+      if (!isUndefined(continuation)) {
+        return c;
+      }
+
+      const projectId = get(c, 'data.id');
+      if (!projectId) {
+        return c;
+      }
+
+      return cache.query(Project, projectId)
+        .then(project => {
+          if (!project) {
+            return c;
+          }
+          return project.$relatedQuery('version')
+            .orderBy('createdAt', 'desc')
+            .first()
+            .select(
+              ref('data:transfer-expiring')
+                .castBool()
+                .as('expiring'),
+              ref('data:project-continuation')
+                .castJson()
+                .as('continuation')
+            );
+        })
```

Comment:
Do we not have the version id on the task, so can query it directly?

Reply:
we do, I just wasn't sure whether to trust it is always kept up to date :)

Reply:
Looking at the task read router and what is used to render the "View submitted version" it looks safe to assume the version id is correct.